### PR TITLE
Support for installation from a branch of StreamSets Data Collector

### PR DESCRIPTION
**Pull request by kirtiv1**
This allows to specify sdc_build and sdc_S3_bucket. Using these two parameters, clusterdock pulls the correct version of SDC and installs it.

### CODE REVIEW

**Review by dimaspivak — CHANGES_REQUESTED** [review]
Pointed Kirti to potentially overhaul how we deal with the SDC parcel (to use a local repo instead of the method I wrote originally). Stay tuned...

File: images/sdc/Dockerfile [hunk 1]
```diff
@@ -13,25 +13,27 @@ FROM alpine:latest
...
-    rm /parcel.tgz
+RUN mkdir -p ${PARCEL_REPO_DIRECTORY} && \
+    wget -P ${PARCEL_REPO_DIRECTORY} ${PARCEL_URL} && \
+    PARCEL_NAME=$(basename ${PARCEL_URL}) && \
+    cd ${PARCEL_REPO_DIRECTORY} && \
+    sha1sum ${PARCEL_NAME} | awk '{ print $1 }' > ${PARCEL_NAME}.sha
 
 # See http://community.cloudera.com/t5/Cloudera-Manager-Installation/Stop-CM-undistributing-my-parcel/m-p/61402.
-RUN touch "${PARCEL_DIRECTORY}/.dont_delete"
+RUN touch "${PARCEL_REPO_DIRECTORY}/.dont_delete"
```
Comment (dimaspivak) [inline images/sdc/Dockerfile]: I think we can remove this (but should do a quick test to be sure)

File: topology.yaml [hunk 2]
```diff
@@ -71,6 +71,9 @@ start args:
     --predictable:
         action: store_true
         help: If specified, attempt to expose container ports to the same port number on the host
-    --sdc-version:
-        help: If specified, the version of StreamSets Data Collector to install
+    --sdc-build:
+        help: If specified, the build from which to install StreamSets Data Collector (e.g. of the form "2038" "3.1.0.0" "latest")
         metavar: ver
+    --sdc-S3-bucket:
```
Comment (dimaspivak) [inline topology.yaml]: `--sdc-s3-bucket`

File: start.py [hunk 3]
```diff
@@ -714,6 +710,23 @@ def _configure_sdc(deployment, cluster, sdc_version, is_kerberos_enabled):
...
+    response = _get(url)
+    if not response.content:
+        logger.info('No SDC version found in manifest.json ...')
+        return None
+    content = response.json()
+    return content['parcels'][0]['components'][0]['version']
+
+
+def _get(url):
+    response = requests.Session().get(url)
```
Comment (dimaspivak) [inline start.py]: No need for `Session`

File: start.py [hunk 4]
```diff
@@ -242,32 +248,22 @@ def cm_server_not_dead(primary_node):
...
-    if args.sdc_version:
+    if args.sdc_build:
+        # We install StreamSets DataCollector using local repo /opt/cloudera/parcel-repo.
+        # Set file and folder permissions correctly.
+        commands = ['chown cloudera-scm:cloudera-scm /opt/cloudera/csd',
+                    'chown cloudera-scm:cloudera-scm /opt/cloudera/parcel-repo',
+                    'chown cloudera-scm:cloudera-scm /opt/cloudera/csd/STREAMSETS*.jar',
+                    'chmod 644 /opt/cloudera/csd/STREAMSETS*.jar',
+                    'chown cloudera-scm:cloudera-scm /opt/cloudera/parcel-repo/STREAMSETS_*']
+        primary_node.execute('; '.join(commands))
```
Comment (dimaspivak) [inline start.py]: Might wanna join with `&&` to make sure failing commands kill it.

**Review by dimaspivak — CHANGES_REQUESTED** [review]
Just a few tiny nits and this is ready to go!

File: start.py (hunk 3 above)
Comment (kirtiv1) [reply start.py]: Done

File: start.py (hunk 4 above)
Comment (kirtiv1) [reply start.py]: Done.

File: images/sdc/Dockerfile (hunk 1 above)
Comment (kirtiv1) [reply images/sdc/Dockerfile]: Done

File: topology.yaml (hunk 2 above)
Comment (kirtiv1) [reply topology.yaml]: Done.

**Comment by srids** [comment]
lgtm +2

**Pull request merged**